<!-- FroboScout launch file -->

<launch>
	<!-- frobit parameters --> 
	<param name="diff_steer_wheel_distance" value="0.215"/>
	<param name="ticks_per_meter_left" value="562" />
	<param name="ticks_per_meter_right" value="562" />

	<!-- node pkg="rosbag" name="rosbag_record" type="record" args="-a"/-->


	<group ns="fmSensors">
	</group>

	<group ns="fmProcessors">
	</group>

	<group ns="fmDecisionMakers">
		<node pkg="sdu_rsd_2014" name="mission" type="keyboard_mission_node.py" output="screen">
			<param name="keyboard_sub" value="/fmHMI/keyboard" />
			<param name="automode_pub" value="/fmDecision/automode" />
    		<param name="deadman_pub" value="/fmCommand/deadman" />
			<param name="cmd_vel_pub" value="/fmCommand/cmd_vel" />
            <param name="max_linear_velocity" value="1.0"/>
            <param name="max_angular_velocity" value="0.8"/>
            <param name="linear_velocity_step" value="0.1"/>
            <param name="angular_velocity_step" value="0.2"/>
		</node>
	</group>

	<group ns="fmExecutors">
		<node pkg="sdu_rsd_2014" type="waypoint_navigation_node.py" name="wptnav" output="screen">
			<param name="automode_sub" value="/fmDecision/automode" />
			<param name="pose_sub" value="/fmKnowledge/pose" />
			<param name="joy_sub" value="/fmLib/joy" />
			<param name="cmd_vel_pub" value="/fmCommand/cmd_vel" />
			<param name="implement_pub" value="/fmCommand/implement" />
			<param name="status_pub" value="/fmInformation/wptnav_status" />
			<param name="pid_pub" value="/fmInformation/wptnav_pid" />
			<param name="status_publish_interval" value="1" />
			<param name="pid_publish_interval" value="1" />
			<param name="print_debug_information" value="false" />

            <param name="max_linear_velocity" value="1.2"/>
            <param name="max_angular_velocity" value="0.70"/>

            <param name="drive_kp" value="2.2"/>
            <param name="drive_ki" value="0.5"/>
            <param name="drive_kd" value="0.8"/>
            <param name="drive_feed_forward" value="0.00"/>
            <param name="drive_max_output" value="0.5"/>

            <param name="turn_kp" value="1.2"/>
            <param name="turn_ki" value="0.8"/>
            <param name="turn_kd" value="0.2"/>
            <param name="turn_feed_forward" value="0.0"/>
            <param name="turn_max_output" value="2.0"/>

			<param name="wpt_default_mode" value="MCTE" />
			<param name="wpt_default_tolerance" value="0.20" />
			<param name="wpt_default_drive_velocity" value="0.4" />
			<param name="wpt_default_turn_velocity" value="1.4" />
			<param name="wpt_default_wait_after_arrival" value="2.0" />
			<param name="wpt_default_implement_command" value="0.0" />

			<param name="target_ahead" value="0.3" />
			<param name="turn_start_at_heading_err" value="60.0" />
			<param name="turn_stop_at_heading_err" value="2.0" />
			<param name="ramp_drive_velocity_at_distance" value="0.25" />
			<param name="ramp_min_drive_velocity" value="0.10" />
			<param name="ramp_turn_velocity_at_angle" value="30.0" />
			<param name="ramp_min_turn_velocity" value="0.3" />
			<param name="stop_navigating_at_distance" value="0.03" />
		</node>		
	</group>

	<group ns="fmControllers">
	</group>

	<group ns="fmLib">
		<node pkg="frobit_sim" name="froboscout_sim" type="frobit_sim_node.py" output="screen" >
    		<param name="deadman_sub" value="/fmCommand/deadman" />
			<param name="cmd_vel_sub" value="/fmCommand/cmd_vel" />
			<param name="odom_reset_sub" value="/fmInformation/odom_reset" />
			<param name="pose_pub" value="/fmKnowledge/pose" />
            <param name="max_linear_acceleration" value="1.0"/>
            <param name="max_angular_acceleration" value="1.0"/>
			<param name="wheel_feedback_left_pub" value="/fmInformation/wheel_feedback_left" />
			<param name="wheel_feedback_right_pub" value="/fmInformation/wheel_feedback_right" />
			<param name="publish_wheel_feedback_rate" value="50" />
		</node>

	 	<node pkg="keyboard_interface" type="keyboard_node" name="keyboard">
			<param name="keyboard_pub" value="/fmHMI/keyboard" />
		</node>

		<node pkg="plot_velocity" name="plot_vel" type="plot_velocity_node.py" output="screen" >
 			<param name="velocity_sub" value="/fmCommand/cmd_vel" />
 			<param name="update_frequency" value="10" />
			<param name="plot_title" value="cmd_vel" />
			<param name="max_linear_velocity" value="1.2" />
			<param name="max_angular_velocity" value="1.0" />
			<param name="samples" value="100" />
		</node>

		<node pkg="plot_propulsion_feedback_leftright" name="plot_feedback" type="plot_propulsion_feedback_leftright_node.py" output="screen" >
 			<param name="propulsion_feedback_left_sub" value="/fmInformation/wheel_feedback_left" />
 			<param name="propulsion_feedback_right_sub" value="/fmInformation/wheel_feedback_right" />
 			<param name="update_frequency" value="50" />
			<param name="plot_title" value="Wheel feedback" />
			<param name="maximum_velocity" value="1.2" />
			<param name="maximum_thrust" value="511" />
		</node>

		<node pkg="plot_pid" name="wptnav_pid" type="plot_pid_node.py" output="screen" >
 			<param name="pid_sub" value="/fmInformation/wptnav_pid" />
 			<param name="update_rate" value="10" />
			<param name="plot_title" value="Navigation PID status" />
			<param name="samples" value="100" />
			<param name="max_error" value="0.5" />
			<param name="max_output" value="0.5" />
			<param name="max_pid" value="1.1" />
			<param name="plot_feed_forward" value="true" />
		</node>

		<node pkg="frobit_plot" name="frobit_plot" type="frobit_plot_node.py" output="screen" >
			<param name="pose_sub" value="/fmKnowledge/pose" />
			<param name="gnss_sub" value="/fmInformation/gpgga_tranmerc" />
			<param name="odom_sub" value="/fmKnowledge/odom" />
			<param name="wptnav_status_sub" value="/fmInformation/wptnav_status" />
			<param name="plot_pose_track" value="true" />
			<param name="plot_gnss_track" value="false" />
			<param name="plot_odometry_track" value="false" />
			<param name="plot_pose_yaw" value="false" />
			<param name="map_trackpoint_threshold" value="0.02" />
			<param name="map_max_trackpoints" value="5000" />
			<param name="map_minimum_size" value="2.0" />
			<param name="map_easting_offset" value="0.0" />
			<param name="map_northing_offset" value="0.0" />
			<param name="map_update_frequency" value="10.0" />
			<param name="map_title" value="Frobit track" />
			<param name="map_window_size" value="6.5" />
			<param name="avatar_extension_front" value="0.29" />
			<param name="avatar_extension_rear" value="0.06" />
			<param name="avatar_extension_lateral" value="0.13" />
			<param name="avatar_reverse" value="false" />
		</node>
	</group>
</launch>

